### PR TITLE
www-server/sniproxy: bump to 0.6.0

### PR DESCRIPTION
**Pull request by nemunaire**
Hi!

Here is a bump for the new release of sniproxy.

I took the opportunity to make some clean and use EAPI 7.

```diff
--- sniproxy-0.5.0.ebuild
+++ sniproxy-0.6.0.ebuild
@@ -1,7 +1,7 @@
-# Copyright 1999-2017 Gentoo Foundation
+# Copyright 1999-2018 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
+EAPI=7
 
 inherit autotools user
 
@@ -60,6 +60,6 @@
 }
 
 pkg_postinst() {
-	enewgroup ${PN}
-	enewuser ${PN} -1 -1 /var/lib/sniproxy ${PN}
+	enewgroup "${PN}"
+	enewuser "${PN}" -1 -1 /var/lib/sniproxy "${PN}"
 }
```

### CODE REVIEW

File: www-servers/sniproxy/sniproxy-0.6.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+LICENSE="BSD-2"
+SLOT="0"
+IUSE="+dns +largefile rfc3339"
+
+RDEPEND="
+	dev-libs/libev
+	>=dev-libs/libpcre-3
+	dns? ( net-libs/udns )
+"
+DEPEND="${RDEPEND}"
```
Comment (gyakovlev) [inline www-servers/sniproxy/sniproxy-0.6.0.ebuild]: I think you are missing a dependency on `sys-devel/gettext` and `virtual/pkgconfig`  here.

Comment (nemunaire) [reply www-servers/sniproxy/sniproxy-0.6.0.ebuild]: Right, I appended them to `DEPEND`.

File: www-servers/sniproxy/sniproxy-0.6.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	SRC_URI="https://github.com/dlundquist/sniproxy/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+DESCRIPTION="Proxies incoming HTTP and TLS connections based on the hostname"
+HOMEPAGE="https://github.com/dlundquist/sniproxy"
+
+LICENSE="BSD-2"
+SLOT="0"
+IUSE="+dns +largefile rfc3339"
```
Comment (gyakovlev) [inline www-servers/sniproxy/sniproxy-0.6.0.ebuild]: seems I missed that on the first review. sorry about that.

what about those useflags? 
I see no `src_configure` that uses them. this does not happen automatically, so you need to explicitly define configure function and pass appropriate flags.

please add something like
```
src_configure() {
        local myconf=(
           #use conditional code here
        )
        econf "${myconf[@]}"
}
```

or drop useflags completely if installing with default options is ok and those options do not change at configure time depending on system.

Comment (nemunaire) [reply www-servers/sniproxy/sniproxy-0.6.0.ebuild]: Oh, I don't know what happen ... I was sure to have already do that for this ebuild...

```diff
--- a/www-servers/sniproxy/sniproxy-0.6.0.ebuild
+++ b/www-servers/sniproxy/sniproxy-0.6.0.ebuild
@@ -41,6 +41,16 @@ src_prepare() {
        sed -i "/create/s/daemon/sniproxy/" debian/logrotate.conf || die "Unable to replace logrotate configuration"
 }

+src_configure() {
+       local my_conf=(
+               $(use_enable dns)
+               $(use_enable largefile)
+               $(use_enable rfc3339 rfc3339-timestamps)
+       )
+
+       econf "${my_conf[@]}"
+}
+
 src_install() {
        default
```

File: www-servers/sniproxy/sniproxy-0.6.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+RDEPEND="
+	dev-libs/libev
+	>=dev-libs/libpcre-3
+	dns? ( net-libs/udns )
+"
+DEPEND="
+	${RDEPEND}
+	sys-devel/gettext
```
Comment (Zlogene) [inline www-servers/sniproxy/sniproxy-0.6.0.ebuild]: these both are BDEPEND in EAPI=7, not DEPEND

**Comment by gyakovlev** [comment]
what is curl doing in depend? if it's needed for tests only then add `test` to `IUSE` and hide it behind `test?` condition.
and squash commits. should be fine after you implement test dep.

**Comment by nemunaire** [comment]
Here it is! I squash the commits to include the bugfix into the bump.
Indeed, as you say, `curl` was only needed for the test part, so I added the `test?` condition.

**Comment by gyakovlev** [comment]
pushed! thanks, and sorry for delay.

**Pull request closed without merging**